relay-general/src/store/transactions/processor.rs

```diff
@@ -396,6 +407,45 @@ impl Processor for TransactionsProcessor<'_> {
         Ok(())
     }
 
+    /// Validates the length and characters of measurement names.
+    fn process_measurements(
+        &mut self,
+        value: &mut crate::protocol::Measurements,
+        _meta: &mut Meta,
+        _state: &ProcessingState<'_>,
+    ) -> ProcessingResult {
+        let measurements = &mut value.0;
+
+        measurements.retain(|key, val| {
+            if !is_valid_metric_name(key) {
+                relay_log::error!("Measurement name contains invalid characters");
+                return false;
+            }
+
+            let name_len = key.len();
+
+            let unit_len = val
+                .value()
+                .and_then(|val| val.unit.value().map(|val| val.to_string().len()));
+
+            if let (Some(fixed_len), Some(max_len), Some(unit_len)) =
+                (self.fixed_metric_len, self.max_metric_name_len, unit_len)
+            {
+                if (name_len + unit_len + fixed_len) > max_len {
+                    relay_log::error!(
+                        "Measurement name is too long({}/{})",
+                        name_len + unit_len + fixed_len,
+                        max_len
+                    );
```

Comment:
See comment about `Meta`.

Reply:
ok, I changed it now and tried to make it similar to your example there, is it now what you have in mind?